tests/_resources/versioned_app/app.py

```diff
@@ -25,6 +25,7 @@ async def lifespan(app: FastAPI):
 versioned_app.add_header_versioned_routers(v2021_01_01_router, header_value="2021-01-01")
 versioned_app.add_header_versioned_routers(v2022_01_02_router, header_value="2022-02-02")
 versioned_app.add_unversioned_routers(webhooks_router)
+versioned_app.enrich_swagger()
```

Comment:
For this case, to satisfy test__lifespan_context_exists(), a new lifespan_app is created, otherwise every time we use custom lifespan, we have to add app.enrich_swagger() to it, everything before yield is startup event, and after shutdown event, which I wanted to avoid. This is also a side effect of approach with add_event_handler("startup", self.enrich_swagger)

What do you think, can it stay like this?